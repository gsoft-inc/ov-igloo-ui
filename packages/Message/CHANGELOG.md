# @igloo-ui/message

## 0.1.0

### Minor Changes

- cf5b505: Initial release
