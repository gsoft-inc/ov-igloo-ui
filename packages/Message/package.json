{
  "name": "@igloo-ui/message",
  "version": "0.0.0",
  "main": "dist/Message.js",
  "module": "dist/Message.js",
  "types": "dist/Message.d.ts",
  "styles": "dist/message.css",
  "private": "true",
  "publishConfig": {
    "access": "public"
  },
  "repository": "git+https://github.com/gsoft-inc/ov-igloo-ui.git",
  "author": "Officevibe Inc.",
  "contributors": [
    "Franck franck.gaudin@gsoft.com"
  ],
  "license": "Apache-2.0",
  "peerDependencies": {
    "react": ">= 16.8.6"
  },
  "files": [
    "dist"
  ],
  "dependencies": {
    "@igloo-ui/icons": "^0.6.6",
    "@igloo-ui/tokens": "^1.0.0",
    "classnames": "^2.3.1"
  },
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "IE >= 11"
  ]
}
