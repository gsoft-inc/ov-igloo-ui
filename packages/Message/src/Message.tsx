import * as React from 'react';
import cx from 'classnames';
import Alert from '@igloo-ui/icons/dist/Alert';

import './message.scss';

export interface MessageProps extends React.ComponentProps<'div'> {
  children: React.ReactNode;
  appearance?: 'default' | 'warning' | 'error';
}

const Message: React.FunctionComponent<MessageProps> = (
  props: MessageProps
) => {
  const { children, appearance = 'default' } = props;
  const classes = cx('ids-message', {
    [`ids-message--${appearance}`]: appearance !== 'default',
  });

  const hasIcon = appearance !== 'default';

  return (
    <span className={classes}>
      {hasIcon && <Alert className="ids-message__icon" size="small" />}
      {children}
    </span>
  );
};

export default Message;
