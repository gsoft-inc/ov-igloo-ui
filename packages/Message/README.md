# @igloo-ui/message

Welcome to my `Message` component.

## Installation

```sh
npm i @igloo-ui/message
# or with yarn
yarn add @igloo-ui/message
```

Then to use the component in your code just import it!

```js
import Message from '@igloo-ui/message';
// and with css-modules
import '@igloo-ui/Message/dist/message.css';
```
