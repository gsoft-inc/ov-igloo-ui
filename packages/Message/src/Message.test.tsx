/**
 * @jest-environment jsdom
 */
import React from 'react';
import { shallow } from 'enzyme';

import Message from './Message';

const setUp = (props = {}) => {
  const component = shallow(<Message {...props}>Information message</Message>);
  return component;
};

describe('Message', () => {
  let component: any;
  beforeEach(() => {
    component = setUp();
  });

  test('It should render without errors', () => {
    const wrapper = component.find('.ids-message');
    expect(wrapper.length).toBe(1);
  });

  test('It should render a default appearance', () => {
    const message = setUp();
    expect(message).toMatchSnapshot();
  });

  test('It should render a error appearance', () => {
    const message = setUp({ appearance: 'error' });
    expect(message).toMatchSnapshot();
  });
});
