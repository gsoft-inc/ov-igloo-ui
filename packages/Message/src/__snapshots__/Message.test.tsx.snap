// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message It should render a default appearance 1`] = `ShallowWrapper {}`;

exports[`Message It should render a error appearance 1`] = `ShallowWrapper {}`;

exports[`Message It should render a waring appearance 1`] = `ShallowWrapper {}`;
